tests/conftest.py

```diff
@@ -271,6 +273,14 @@ def random_user_uttered_event(timestamp: Optional[float] = None) -> UserUttered:
     )
 
 
+def pytest_runtest_setup(item: pytest.Item) -> None:
+    supported_platforms = PLATFORMS.intersection(
+        mark.name for mark in item.iter_markers()
+    )
+    if supported_platforms and sys.platform not in supported_platforms:
+        pytest.skip(f"cannot run on platform {sys.platform}")
```

Comment:
as @wochinge suggested, could we switch the decorators around and mark them as `skip_on_windows`?

Reply:
it will make it much harder to use these decorators with `pytest.mark.parametrize`. For example, this is what platform-specific decorators allow us to do:
```
@pytest.mark.parametrize(
    "_config",
    [
        {"pipeline": [{"name": "WhitespaceTokenizer"}, {"name": "SpacyFeaturizer"}]},
        pytest.param(
            {
                "pipeline": [
                    {"name": "WhitespaceTokenizer"},
                    {"name": "ConveRTFeaturizer"},
                ]
            },
            marks=[pytest.mark.linux, pytest.mark.darwin],
        ),
        pytest.param(
            {
                "pipeline": [
                    {"name": "ConveRTTokenizer"},
                    {"name": "LanguageModelFeaturizer"},
                ]
            },
            marks=[pytest.mark.linux, pytest.mark.darwin],
        ),
    ],
)
```
There is no way to do the same thing if you mark tests with `skip_on_windows`. It can be something like `unix-only` but I personally find it less clear than just specifying platforms directly.

Reply:
but we're only using `marks=` for two tests right?
1. `test_missing_required_component()`: both versions of this parametrized test are marked `[pytest.mark.linux, pytest.mark.darwin]`, so we could just mark the whole test as `mark.skip_on_windows`
2. `test_train_persist_load_with_different_settings()` - one part of this test runs on windows, the other one doesn't. is it possible that the `skip_on_windows` marker would still work even inside the `marks=` kwarg? if not, we could still just break up this test into two, and not parametrize (one for linux, one for windows)